core/services/network_checker.rb

```diff
@@ -98,4 +93,15 @@ def self.check_available_tool?(tool, machine_configurator, machine, logger)
     logger.debug("#{tool} is not available")
     false
   end
+
+  # Prints table and return results of the network resource availability check
+  def self.parse_table(table, logger)
```

Comment:
I have previously stated that you must do the printing of the table and checking the results in separate actions, not in one. Split this method in two.